PyMca5/PyMcaCore/TiledDataSource.py

```diff
@@ -0,0 +1,136 @@
+import os
+import time
+
+from PyMca5.PyMcaIO import TiledFile
+
+SOURCE_TYPE = 'Tiled'
+
+_logger = logging.getLogger(__name__)
+
+class TiledDataSource(object):
+    """
+    Creates instance of a Tiled Data Source class. This is neccesary
+    to create a Tiled Tab in the QDispatcher, which houses the Tiled
+    Browser.
+
+    This is largely based on the NexusDataSource class, but all Data 
+    Source tabs (Spec, EDF, SPS) have an analogous class.
+
+    See QDataSource.py
+    """
+
+    def __init__(self, nameInput):
+        self.nameList = nameInput
+        self.source_type = SOURCE_TYPE
+        self.__sourceNameList = self.__sourceNameList
+        self._sourceObjectList = []
+        self.refresh()
+
+    def refresh(self):
+        pass
+
+    def getSourceInfo(self):
+        """
+        Returns a dictionary with the key "KeyList" (list of all available keys
+        in this source). Each element in "KeyList" has the form 'n1.n2' where
+        n1 is the source number and n2 entry number in file both starting at 1.
+        """
+        return self.__getSourceInfo()
+    
+    def __getSourceInfo(self):
+        SourceInfo={}
+        SourceInfo["SourceType"]=SOURCE_TYPE
+        SourceInfo["KeyList"]=[]
+        i = 0
+        for sourceObject in self._sourceObjectList:
+            i+=1
+            nEntries = len(sourceObject["/"].keys())
+            for n in range(nEntries):
+                SourceInfo["KeyList"].append("%d.%d" % (i,n+1))
+        SourceInfo["Size"]=len(SourceInfo["KeyList"])
+        return SourceInfo
+    
+    def getKeyInfo(self, key):
+        if key in self.getSourceInfo()['KeyList']:
+            return self.__getKeyInfo(key)
+        else:
+            #should we raise a KeyError?
+            _logger.debug("Error key not in list ")
+            return {}
+
+    def __getKeyInfo(self,key):
+        try:
+            index, entry = key.split(".")
+            index = int(index)-1
+            entry = int(entry)-1
+        except Exception:
+            #should we rise an error?
+            _logger.debug("Error trying to interpret key = %s", key)
+            return {}
+        
+        sourceObject = self._sourceObjectList[index]
+        info = {}
+        info["SourceType"]  = SOURCE_TYPE
+        #doubts about if refer to the list or to the individual file
+        info["SourceName"]  = self.sourceName[index]
+        info["Key"]         = key
+        #specific info of interest
+        info['FileName'] = sourceObject.name
+        return info
+    
+    def getDataObject(self):
+        pass
+
+    def isUpdated(self, sourceName, key):
+        #sourceName is redundant?
+        index, entry = key.split(".")
+        index = int(index)-1
+        lastmodified = os.path.getmtime(self.__sourceNameList[index])
+        if lastmodified != self.__lastKeyInfo[key]:
+            self.__lastKeyInfo[key] = lastmodified
+            return True
+        else:
+            return False
```

Comment:
I realize that this is coming from mostly from the NexusDataSource example.  It looks like this will need to get modified with the results coming out of the `QTiledWidget` if I'm following it correctly.